src/wcm.cpp

```diff
@@ -338,28 +339,124 @@ OptionWidget::OptionWidget(Option *option) : Gtk::Box(Gtk::ORIENTATION_HORIZONTA
     switch (option->type)
     {
       case OPTION_TYPE_INT:
+      case OPTION_TYPE_ANIMATION:
     {
-        auto value_optional = wf::option_type::from_string<int>(
-            wf_option->get_value_str());
-        int value = value_optional ? value_optional.value() : std::get<int>(
-            option->default_value);
+        int value;
+        int default_value;
+        std::string easing_type;
+        if (option->type == OPTION_TYPE_INT)
+        {
+            value = wf::option_type::from_string<int>(wf_option->get_value_str()).value_or(
+                std::get<int>(option->default_value));
+            default_value = std::get<int>(option->default_value);
+        } else // OPTION_TYPE_ANIMATION
+        {
+            auto set_value =
+                wf::option_type::from_string<wf::animation_description_t>(wf_option->get_value_str());
+            value = set_value.has_value() ? set_value->length_ms : std::get<int>(option->default_value);
+            default_value =
+                wf::option_type::from_string<wf::animation_description_t>(std::get<std::string>(option->
+                    default_value))->length_ms;
+            easing_type = wf::option_type::from_string<wf::animation_description_t>(
+                wf_option->get_value_str())->easing_name;
+        }
 
         if (option->int_labels.empty())
         {
             auto spin_button = std::make_unique<Gtk::SpinButton>(
                 Gtk::Adjustment::create(value, option->data.min, option->data.max,
                     1));
             spin_button->signal_value_changed().connect(
-                [=, widget = spin_button.get()]
+                [=, widget = spin_button.get(), easing_type = easing_type]
                 {
-                    option->set_save(widget->get_value_as_int());
-                });
-            reset_button.signal_clicked().connect(
-                [=, widget = spin_button.get()]
-                {
-                    widget->set_value(std::get<int>(option->default_value));
+                    if (option->type == OPTION_TYPE_INT)
+                    {
+                        option->set_save(widget->get_value_as_int());
+                    } else
+                    {
+                        auto set_value =
+                            wf::option_type::from_string<wf::animation_description_t>(
+                                wf_option->get_value_str());
+                        auto easing =
+                            set_value.has_value() ? set_value->easing_name : std::get<std::string>(option->
+                                default_value);
+                        std::string animation_value = std::to_string(
+                            widget->get_value_as_int()) + "ms " + easing;
```

Comment:
Maaaybe a better approach would be to add a `Option::set_value` overload which takes two parameters (`int` and `std::string`). It would simplify code here to `option->set_save(default_value, default_easing_type)` and allow to avoid code duplication (here and below in reset and `combo_box->signal_changed()` callbacks),

Reply:
I'll leave this for you to do.